install-papirus-home.sh

```diff
@@ -1,15 +1,60 @@
-#/bin/bash
+#!/bin/bash
+
+# Download and install the latest Papirus Icon theme
+
+#set -x
+set -e
+
+DL_URL="https://github.com/PapirusDevelopmentTeam/papirus-icon-theme-gtk/archive/master.zip"
+TARGET_DIR=$HOME/.icons
+HASH_FILE=$TARGET_DIR/.papirus-icons.etag
+
+get_remote_hash() {
+    curl -i -s -I -L $DL_URL | grep -i '^ETag: '
+}
+
+get_local_hash() {
+    touch $HASH_FILE
+    cat $HASH_FILE
+}
+
+extract() {
+    if [ -x /usr/bin/7z ]; then
+        /usr/bin/7z x "$1" -o"$2"
+    else
+        /usr/bin/unzip "$1" -d "$2"
+    fi
+}
+
 echo "Papirus icon theme for GTK"
-sleep 5
-echo "Delete old Papirus icon theme ..."
-rm -rf ~/.icons/{Papirus-GTK,Papirus-Dark-GTK}
+
+mkdir -p $TARGET_DIR
+
+echo "Comparing versions ..."
+REMOTE_HASH=$(get_remote_hash)
+LOCAL_HASH=$(get_local_hash)
+
+if [ "$REMOTE_HASH" == "$LOCAL_HASH" ]
+then
+    echo "Already using the latest version"
+    exit 0
+fi
+
 echo "Download new version from GitHub ..."
-wget -c https://github.com/PapirusDevelopmentTeam/papirus-icon-theme-gtk/archive/master.zip -O /tmp/papirus-icon-theme-gtk.zip
+TEMP_DIR=$(mktemp -d)
+curl -L $DL_URL -o $TEMP_DIR/master.zip
+
 echo "Unpack archive ..."
-7z x /tmp/papirus-icon-theme-gtk.zip -o/tmp/
+extract $TEMP_DIR/master.zip $TEMP_DIR
+
+echo "Delete old Papirus icon theme ..."
+rm -rf $TARGET/{Papirus-GTK,Papirus-Dark-GTK}
+
 echo "Installing ..."
-mkdir ~/.icons
```

Comment:
I think this line doesn't need to be removed. The first time I run the home installation script I didn't have any complains about the ~/.icons directory already existing but I had the complains the second time I run that script which makes me think that I didn't have that directory initially.

I think that you should change this line for: mkdir -p ~/.icons to avoid any complain and ensure that the directory exists before executing the mv.